src/lib/codemirror/init.ts

```diff
@@ -1,30 +1,71 @@
-import { EditorState } from '@codemirror/state'
-import { getSearchQuery, highlightSelectionMatches, search, searchKeymap, setSearchQuery } from '@codemirror/search'
-import widgets from './widgets'
-import { effect, signal } from '@preact/signals'
-import { h, render } from 'preact'
-import { bracketMatching, defaultHighlightStyle, foldGutter, foldKeymap, indentOnInput, indentUnit, syntaxHighlighting } from '@codemirror/language'
-import { history, defaultKeymap, historyKeymap, indentWithTab, insertNewlineAndIndent } from '@codemirror/commands'
-import { javascript } from '@codemirror/lang-javascript'
-import SearchBox from '../../components/search-box'
-import { EditorView, keymap, lineNumbers, highlightActiveLineGutter, highlightSpecialChars, drawSelection, dropCursor, rectangularSelection, crosshairCursor, highlightActiveLine } from '@codemirror/view'
+import { autocompletion } from "@codemirror/autocomplete";
+import {
+	defaultKeymap,
+	history,
+	historyKeymap,
+	indentWithTab,
+	insertNewlineAndIndent,
+} from "@codemirror/commands";
+import { javascript } from "@codemirror/lang-javascript";
+import {
+	bracketMatching,
+	defaultHighlightStyle,
+	foldGutter,
+	foldKeymap,
+	indentOnInput,
+	indentUnit,
+	syntaxHighlighting,
+} from "@codemirror/language";
 import { lintGutter } from "@codemirror/lint";
-import type { NormalizedError } from '../state'
+import {
+	getSearchQuery,
+	highlightSelectionMatches,
+	search,
+	searchKeymap,
+	setSearchQuery,
+} from "@codemirror/search";
+import { EditorState } from "@codemirror/state";
+import {
+	crosshairCursor,
+	drawSelection,
+	dropCursor,
+	EditorView,
+	highlightActiveLine,
+	highlightActiveLineGutter,
+	highlightSpecialChars,
+	keymap,
+	lineNumbers,
+	rectangularSelection,
+} from "@codemirror/view";
+import { effect, signal } from "@preact/signals";
+import { h, render } from "preact";
+import SearchBox from "../../components/search-box";
+import type { NormalizedError } from "../state";
+import { sprigFunctionCompletion } from "./autocomplete";
+import widgets from "./widgets";
```

Comment:
It's generally good to avoid formatting changes in a PR as it makes it hard to read the diff. Can you undo the formatting here?